product-web-app/service/authService.ts

```diff
@@ -37,13 +38,19 @@ export default class AuthService {
     }
   }
 
-  async login(user: User): Promise<void> {
+  async login(user: User, res: Response): Promise<void> {
```

Comment:
Move setting a cookie to the controller, this method can just return extracted cookie from a body. You won't require those tests using sinon